aiidalab_widgets_base/process.py

```diff
@@ -111,30 +125,92 @@ def on_submitted(self, function):
         self._run_after_submitted.append(function)
 
 
-class ProcessFollowerWidget(VBox):
+class ProcessInputsWidget(ipw.VBox):
+    """Widget to select and show process inputs."""
+
+    def __init__(self, process=None, **kwargs):
+        self.process = process
+        self.output = ipw.Output()
+        self.info = ipw.HTML()
+        inputs_list = [(l.title(), l) for l in self.process.inputs] if self.process else []
+        inputs = ipw.Dropdown(
+            options=[('Select input', '')] + inputs_list,
+            description='Select input:',
+            style={'description_width': 'initial'},
+            disabled=False,
+        )
+        inputs.observe(self.view_input, names=['value'])
+        super().__init__(children=[ipw.HBox([inputs, self.info]), self.output], **kwargs)
+
+    def view_input(self, change=None):
+        """Show selected input."""
+        with self.output:
+            self.info.value = ''
+            clear_output()
+            if change['new']:
+                selected_input = self.process.inputs[change['new']]
+                self.info.value = "PK: {}".format(selected_input.id)
+                display(viewer(selected_input))
+
+
+class ProcessOutputsWidget(ipw.VBox):
+    """Widget to select and show process outputs."""
+
+    def __init__(self, process=None, **kwargs):
+        self.process = process
+        self.output = ipw.Output()
+        self.info = ipw.HTML()
+        outputs_list = [(l.title(), l) for l in self.process.outputs] if self.process else []
+        outputs = ipw.Dropdown(
+            options=[('Select output', '')] + outputs_list,
+            label='Select output',
+            description='Select outputs:',
+            style={'description_width': 'initial'},
+            disabled=False,
+        )
+        outputs.observe(self.view_output, names=['value'])
+        super().__init__(children=[ipw.HBox([outputs, self.info]), self.output], **kwargs)
+
+    def view_output(self, change=None):
```

Comment:
Ah, I forgot one thing here. I wanted to make a base class for `ProcessInputsWidget` and `ProcessOutputsWidget` classes since code is mostly the same. Do you think it is a good idea?

Reply:
Changed for both: `ProcessInputsWidget` and `ProcessOutputsWidget`